lib/src/database/matrix_sdk_database.dart

```diff
@@ -693,14 +694,10 @@ class MatrixSdkDatabase extends DatabaseApi {
                 'user_id': userId,
                 'outdated': deviceKeysOutdated[userId],
               },
-              childEntries
-                  .where((c) => c != null)
-                  .toList()
-                  .cast<Map<String, dynamic>>(),
-              crossSigningEntries
-                  .where((c) => c != null)
-                  .toList()
-                  .cast<Map<String, dynamic>>(),
+              List<Map<String, dynamic>>.from(
+                  childEntries.whereType<Map<String, dynamic>>()),
+              List<Map<String, dynamic>>.from(
+                  crossSigningEntries.whereType<Map<String, dynamic>>()),
```

Comment:
`List<T>.from()` is not better than `list.cast<T>()` as it crashes in the same way when there is an entry which is not `T`. The difference is that `List<T>.from()` create a non-deep copy. Do we need this here? Also this is unrelated to the bug described in the commit message